jl4-cli/tests/Main.hs

```diff
@@ -55,11 +61,18 @@ l4Golden dir inputFile = do
 
 jl4ExactPrintGolden :: String -> String -> IO (Golden Text)
 jl4ExactPrintGolden dir inputFile = do
-  input <- Text.readFile inputFile
-  let output_ = JL4.exactprintProgram (takeFileName inputFile) input
+  (errs, moutput) <- oneshotL4ActionAndErrors inputFile \nfp -> do
+    let uri = normalizedFilePathToUri nfp
+    Shake.addVirtualFileFromFS nfp
+    Shake.use Rules.ExactPrint uri
+
+  let regex = Text.pack "<<file>>" <$ (
+        many (RE.satisfy isSpace) *> RE.text "file://" *> some (RE.satisfy (not . isSpace)))
```

Comment:
Ok, as remarked above, I can see why you'd want to remove / reformat the URI, but it should be easy enough to get the basename back in?

Reply:
so the name of the file itself? sure.